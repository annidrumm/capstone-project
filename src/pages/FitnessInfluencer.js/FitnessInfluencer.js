import React from 'react'
import influencerData from '../../mocks/influencer.json'
import InfluencerList from '../InfluencerList/InfluencerList'

export default function FitnessInfluencer() {
  const filteredCategories = influencerData.filter(function (profile) {
    return profile.categories.includes('Fitness')
  })

  return <InfluencerList influencerData={filteredCategories} />
}
