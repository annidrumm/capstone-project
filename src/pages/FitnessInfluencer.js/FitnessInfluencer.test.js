import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import FitnessInfluencer from './FitnessInfluencer.js'
import 'jest-styled-components'

describe('InfluencerList', () => {
  const influencerData = [
    {
      id: 1,
      username: '@fitness4life',
      categories: ['Fitness'],
      follower: 10000,
      gender: 'female',
      age: '25 years',
      location: 'Hamburg',
      languages: ['German', 'English'],
      image:
        'https://raw.githubusercontent.com/annidrumm/influencer-finder/filter-categories/src/images/wesley-tingey-min.jpg',
    },
  ]

  it('display the username from a influencer', () => {
    const { getByText, getAllByText } = render(<FitnessInfluencer />)
    influencerData.forEach(
      ({
        username,
        follower,
        location,
        age,
        gender,
        languages,
        categories,
      }) => {
        expect(getByText(username)).toBeInTheDocument()
        expect(getByText(new RegExp(age, 'i'))).toBeInTheDocument()
        expect(getByText(new RegExp(location, 'i'))).toBeInTheDocument()
        expect(getAllByText(new RegExp(gender, 'i'))[0]).toBeInTheDocument()
        expect(JSON.stringify(new RegExp(languages, 'i'))).toBe(
          JSON.stringify(/languages/i)
        )
        expect(JSON.stringify(new RegExp(categories, 'i'))).toBe(
          JSON.stringify(/categories/i)
        )
      }
    )
  })

  it('renders username correctly', () => {
    const tree = renderer.create(<FitnessInfluencer />)
    expect(tree).toMatchSnapshot()
  })
})
