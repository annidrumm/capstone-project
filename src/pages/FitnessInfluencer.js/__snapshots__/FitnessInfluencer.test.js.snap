// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfluencerList renders username correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-column: 1/3;
  margin: 20px 0;
  text-align: center;
}

.c2 {
  display: inline;
  position: relative;
  font: var(--font-weight-regular);
  font-size: 25px;
  padding-bottom: 5px;
  margin: auto;
  max-width: 300px;
}

.c0 {
  position: absolute;
  left: 30px;
  margin-top: 10px;
  cursor: pointer;
}

.c3 {
  background: var(--transparent-blue);
  height: 350px;
}

.c4 {
  width: 150px;
  height: 220px;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 20px 0 0 0;
  border-radius: 50%;
}

.c5 {
  font-family: 'Poppins';
  font-weight: var(--font-weight-semibold);
  text-align: center;
  padding: 10px 0 0 0;
  margin-top: 0;
  margin-bottom: 0;
}

.c6 {
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  text-align: center;
  margin-top: 0;
  margin-bottom: 0;
}

.c7 {
  text-align: left;
  margin: 0;
  list-style-type: none;
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  margin: 50px 0 120px 0;
  width: 300px;
  margin-left: auto;
  margin-right: auto;
}

.c8 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin: 10px 0;
  position: relative;
}

<section
  className="profiles"
>
  <img
    alt="arrow left"
    className="c0"
    onClick={[Function]}
    src="back.svg"
  />
  <section
    className="c1"
  >
    <h1
      className="c2"
    >
      Search
    </h1>
  </section>
  <section>
    <section
      className="c3"
    >
      <img
        className="c4"
        src="https://raw.githubusercontent.com/annidrumm/influencer-finder/filter-categories/src/images/wesley-tingey-min.jpg"
      />
      <h2
        className="c5"
      >
        @fitness4life
      </h2>
      <h3
        className="c6"
      >
        Fitness
      </h3>
    </section>
    <ul
      className="c7"
    >
      <li
        className="c8"
      >
        Follower: 
        10,000
      </li>
      <li
        className="c8"
      >
        Gender: 
        female
      </li>
      <li
        className="c8"
      >
        Age: 
        25 years
      </li>
      <li
        className="c8"
      >
        Location: 
        Hamburg
      </li>
      <li
        className="c8"
      >
        Languages: 
        German, English
      </li>
    </ul>
  </section>
</section>
`;
